static/js/modules/breakdown-map.js

```diff
@@ -0,0 +1,165 @@
+  this.$elm.find('.js-category').on('change', this.handleCategoryChange.bind(this));
+}
+
+BreakdownMap.prototype.init = function() {
+  this.basePath = ['schedules', 'schedule_a', 'by_state', 'totals'];
+
+  this.baseQuery = {
+    committee_type: 'ALL',
+    per_page: 100,
+    sort_hide_null: true,
+    cycle: this.cycle,
+    sort: '-total'
+  };
+
+  this.loadData(this.baseQuery);
+};
+
+BreakdownMap.prototype.handleCycleChange = function(e) {
+  e.preventDefault();
+  this.baseQuery.cycle = e.target.value;
+  this.destroyTableAndMap();
+  this.loadData(this.baseQuery);
+};
+
+BreakdownMap.prototype.handleCategoryChange = function(e) {
+  this.baseQuery.committee_type = e.target.value;
+  this.destroyTableAndMap();
+  this.loadData(this.baseQuery);
+};
+
+BreakdownMap.prototype.loadData = function(query) {
+  var self = this;
+
+  $.getJSON(
+    helpers.buildUrl(this.basePath, query)
+  ).done(function(response) {
+    self.buildMap(response);
+  });
+
+  this.buildTable();
+};
+
+BreakdownMap.prototype.buildMap = function(data) {
+  var self = this;
+  var opts = {
+    width: 350,
+    height: 300,
+    scale: 400,
+    translate: [160, 120],
+    min: null,
+    max: null,
+    addLegend: true,
+    addTooltips: true,
+  };
+  maps.stateMap(this.$map, data, opts);
+
+  events.on('state.table', function(params) {
+    self.highlightRowAndState(params.state, false);
+  });
+
+  this.$map.on('click', 'path[data-state]', function() {
+    var state = $(this).attr('data-state');
+    events.emit('state.map', {state: state});
+  });
+};
+
+BreakdownMap.prototype.buildTable = function() {
+  var self = this;
+
+  this.stateTable = new tables.DataTable(this.$table, {
+    path: this.basePath,
+    query: this.baseQuery,
+    columns: stateColumns,
+    dom: 't',
+    ordering: false,
+    paging: false,
+    scrollY: 400,
+    scrollCollapse: true
+  });
+  events.on('state.map', function(params) {
+    self.highlightRowAndState(params.state, true);
+  });
+  this.$table.on('click', 'tr', function() {
+    events.emit('state.table', {
+      state: $(this).find('span[data-state]').attr('data-state')
+    });
+  });
+};
+
+BreakdownMap.prototype.highlightRowAndState = function(state, scroll) {
+  var $scrollBody = this.$table.closest('.dataTables_scrollBody');
+  var $row = $scrollBody.find('span[data-state="' + state + '"]');
+
+  if ($row.length > 0) {
+    maps.highlightState($('.state-map'), state);
+    $scrollBody.find('.row-active').removeClass('row-active');
+    $row.parents('tr').addClass('row-active');
+    if (scroll) {
+      $scrollBody.animate({
+        scrollTop: $row.closest('tr').height() * parseInt($row.attr('data-row'))
+      }, 500);
+    }
+  }
+};
+
+BreakdownMap.prototype.destroyTableAndMap = function() {
+  this.stateTable.destroy();
+  this.$map.find('svg').remove();
+  this.$map.find('.legend-container').append('<svg></svg>');
+};
+
+BreakdownMap.prototype.updateDates = function() {
+  var today = new Date();
```

Comment:
Is this code shared with the `TopEntities` component? If so maybe should be generalized into a helper method?